### ISSUE TITLE
Deleting the original blog post causes issues.

### ISSUE THREAD

**Issue by jprochazka** [jprochazka/adsb-receiver#117]
Issue as reported on the FlightAware forums:
http://discussions.flightaware.com/post190105.html#p190105

> Also I noticed a problem when creating a new blog front page.
> I did this by creating a new blog, then deleting the original default one.
> when I did this, the web server front page did not work.
> I installed the web server element again using ./install.sh and things were ok again with the default blog showing as the front page.
> The only way I could get a custom blog front page to work was to edit the original one dated 27 Jan 16 , ok as a workaround but the date of the original default blog cannot be changed.

**Comment by jprochazka** [comment]
Cannot recreate the issue. Waiting on response back from the person who reported it.

**Comment by UlrichZeller** [comment]
I had this problem too.
Created new post -> OK
Deleteted original post (27 Jan) -> OK
Created second new post. Publish -> Error
Unfortunately I can not reproduce this.

**Comment by UlrichZeller** [comment]
Deleted original post -> OK
Created first new post -> OK
Created second new post. Publish -> Error: Cannot create blogPosts object

Attached /var/www/html/data/blogPosts.xml
[blogPosts.xml.zip](https://github.com/jprochazka/adsb-receiver/files/176607/blogPosts.xml.zip)

**Comment by jprochazka** [comment]
Going to put this one on hold until after the 2.0.0 release. A lot of changes to the way data is accessed were just added and will be merged once this branch is completed.

**Comment by radiostationx** [comment]
Thought I should add,
There doesnt seem to be a way to copy txt from wordpad/word/notepad document and paste to the blog editor window. There is a paste selection in the blog editor but the txt selected and copied from another source will not paste. Almost as if the original selection I wanted to transfer over to blog wasnt copied.
Sorry the description of the problem is sketchy.
Workaround is still edit the original blog page and save but if you should run ./install.sh to update to the lates additions the blog details are lost.

**Comment by jprochazka** [comment]
Looking for feedback on this issue using the v2.0.0 release. If nothing is reported I will consider this issue resolved.

**Comment by radiostationx** [comment]
Confirmed that writing blog, saving and editing existing txt all fine in v2.0.0.

On editing the existing blog, Im Still trying to find out how i can highlight text from other sources for example .doc or a random webpage and paste the text into blog window.
This would be useful to prepare blog revisions using a word processor then pasting the content into the blog window and save.

**Comment by hasselk** [comment]
this behaviour seems to be due to errors with parsing &nbsp;, if i delete those the page works again

it seems to be ok now after the update to the actual repo.

**Comment by jprochazka** [comment]
Since &nbsp; is not defined as an XML entity we need to replace it with it's numeric character reference &#160;. This was fixed by replacing &nbsp; with &#160; before storing an new or edited posts in blogPosts.xml. This is not needed for nor done to data stored in MySQL or SQLite.

Commit 5e3374b7fbb90f068b551eed0ea47375ae734e04 contains this change.
